internal/core/operations/remove_rooms/remove_rooms_executor_test.go

```diff
@@ -0,0 +1,121 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+//+build unit
+
+package remove_rooms
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	clock_mock "github.com/topfreegames/maestro/internal/adapters/clock/mock"
+	instance_storage_mock "github.com/topfreegames/maestro/internal/adapters/instance_storage/mock"
+	port_allocator_mock "github.com/topfreegames/maestro/internal/adapters/port_allocator/mock"
+	"github.com/topfreegames/maestro/internal/adapters/room_storage/mock"
+	runtime_mock "github.com/topfreegames/maestro/internal/adapters/runtime/mock"
+
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	porterrors "github.com/topfreegames/maestro/internal/core/ports/errors"
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+)
+
+func TestExecute(t *testing.T) {
```

Comment:
@gabrielcorado missing test case when `ListRoomsWithDeletionPriority` method returns with an error, what do you think about including this one here?